sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -687,7 +687,8 @@ class Analyzer(
         resolved
       } else {
         plan match {
-          case u: UnaryNode if !u.isInstanceOf[SubqueryAlias] =>
+          case u: UnaryNode
```

Comment:
There are 31 `UnaryNode` types. Actually, I did check most of `UnaryNode`. 

`ResolveSortReferences` resolving missing attributes if and only if their child nodes have been resolved, as shown below. 
```scala
case s @ Sort(order, _, child) if !s.resolved && child.resolved 
```

Thus, most of ineligible `UnaryNode` have been replaced. For example, `GroupingSets` and `Pivot` will not be seen in this resolution. Thus, ignoring them is fine. Besides `Subquery`, I think `ScriptTransformation`, `MapPartitions`, `AppendColumns` and `MapGroups` need to be excluded. 

Or should we use a white list here?